### PR TITLE
Do not remove serif font family in pgf example to avoid Warning

### LINKED ISSUE #20850: [Bug]: Warning: Font Family not found
### Bug summary

Usign Latex fonts as recommended here throws a Warning:

```
findfont: Font family ['serif'] not found. Falling back to DejaVu Sans.
findfont: Generic family 'serif' not found because none of the following families were found: 
```

### Code for reproduction

```python
import matplotlib.pyplot as plt

plt.rcParams.update({
    "font.family": "serif",
    # Use LaTeX default serif font.
    "font.serif": [],
    # Use specific cursive fonts.
    "font.cursive": ["Comic Neue", "Comic Sans MS"],
})

plt.figure()
plt.plot([1,2,3,4])
plt.savefig('test.png')
plt.savefig('test.pgf')
```

### Actual outcome

The Bug is also visible in your Docs.
https://matplotlib.org/stable/gallery/userdemo/pgf_fonts.html

### Expected outcome

No Warning should get thrown everytime

### Operating system

Ubuntu

### Matplotlib Version

3.4.2

### Matplotlib Backend

TkAgg

### Python version

3.9.5

### Jupyter version

_No response_

### Other libraries

_No response_

### Installation

pip

### Conda channel

_No response_

### PR DESCRIPTION
**Pull request by FranzForstmayr**
Currently the pgf_plots example throws a warning that the pgf font serif is not found.
This PR fixes this behaviour.

Fixes #20850

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by aitikgupta — APPROVED** [review]
I'm not sure why there's no "Check the rendered docs here!" GitHub Action check for this PR.. but this looks good, thanks!

**Comment by QuLogic** [comment]
Because Circle hasn't run a doc CI build at all.

**Comment by aitikgupta** [comment]
There seems to be a separate bug; that the Comic font isn't found by CI:
https://63196-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/userdemo/pgf_fonts.html

This is true for other PRs too (for example, the most recent https://github.com/matplotlib/matplotlib/pull/20949): https://63199-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/userdemo/pgf_fonts.html

It did use to find the cursive fonts for the previous builds (on stable): https://matplotlib.org/stable/gallery/userdemo/pgf_fonts.html

**Comment by QuLogic** [comment]
`/stable/` is built on my machine, not CI. You can't compare it to `/devdocs/` or PR builds.

**Comment by QuLogic** [comment]
It appears that we use Comic (Neue, Sans MS) because it's already listed in `font.cursive`, but it's not available on CI. Felipa is also listed in that font list, and is explicitly installed on CI, so maybe we should switch to that instead?

**Comment by anntzer** [comment]
As discussed during the dev call this is actually hiding a bug that I bisected back to #10339; the fix is clear looking at that PR, and should be something like
```patch
diff --git i/lib/matplotlib/backends/backend_pgf.py w/lib/matplotlib/backends/backend_pgf.py
index 749da55966..8706fed514 100644
--- i/lib/matplotlib/backends/backend_pgf.py
+++ w/lib/matplotlib/backends/backend_pgf.py
@@ -50,9 +50,14 @@ def get_fontspec():
         for family, command in zip(families, commands):
             # 1) Forward slashes also work on Windows, so don't mess with
             # backslashes.  2) The dirname needs to include a separator.
-            path = pathlib.Path(fm.findfont(family))
-            latex_fontspec.append(r"\%s{%s}[Path=\detokenize{%s}]" % (
-                command, path.name, path.parent.as_posix() + "/"))
+            try:
+                path = pathlib.Path(
+                    fm.findfont(family, fallback_to_default=False))
+            except ValueError:
+                pass
+            else:
+                latex_fontspec.append(r"\%s{%s}[Path=\detokenize{%s}]" % (
+                    command, path.name, path.parent.as_posix() + "/"))
 
     return "\n".join(latex_fontspec)
```

**Review by anntzer — CHANGES_REQUESTED** [review]
Per the above.

**Comment by jklymak** [comment]
I'm moving to draft until the above is addressed.  Thanks!

File: examples/userdemo/pgf_fonts.py [hunk 1]
```diff
@@ -7,9 +7,9 @@
 
 import matplotlib.pyplot as plt
 plt.rcParams.update({
+    # Use LaTeX default serif font-family.
+    # https://matplotlib.org/stable/tutorials/text/usetex.html#sphx-glr-tutorials-text-usetex-py
     "font.family": "serif",
-    # Use LaTeX default serif font.
-    "font.serif": [],
```
Comment (anntzer) [inline examples/userdemo/pgf_fonts.py:12]: this should not be changed anymore?

**Comment by anntzer** [comment]
I think this just needs the change to the example to be reverted, but otherwise looks good.

File: examples/userdemo/pgf_fonts.py (hunk 1 above)
Comment (aitikgupta) [reply examples/userdemo/pgf_fonts.py:12]: @FranzForstmayr can you do this small revert? Basically an empty list is valid (and now fixed) for PGF backend.

**Comment by FranzForstmayr** [comment]
Reverting the change does not seem to work. 
https://63832-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/userdemo/pgf_fonts.html

I can remove the empty list from the font specification again, otherwise I'm not sure how to fix this error.

**Comment by anntzer** [comment]
Actually there's another problem with the example, which is that it doesn't actulaly use the pgf backend (which is a pity...)
One needs to add e.g. `backend="pgf"` to each of the savefig calls.
That then shows another problem with the example, which is that `family="cursive"` doesn't work, which I *guess* could just be documented as a limitation of backend_pgf (because TeX has `\rmfamily \sffamily \ttfamily` but no `\cursivefamily` (or `\calfamily`, I guess).

**Comment by jklymak** [comment]
I popped this back to draft, but feel free to put back on the queue

**Comment by QuLogic** [comment]
Hmm, @FranzForstmayr did you intend to close this?

**Comment by FranzForstmayr** [comment]
Yes, as there is a more recent PR (#22111) about this topic.

**Pull request closed without merging**